cpu/kinetis_common/i2c.c

```diff
@@ -227,18 +227,14 @@ static inline int _i2c_receive(I2C_Type *dev, uint8_t *data, int length)
 
         dev->S = I2C_S_IICIF_MASK;
 
-        if (length == 2) {
-            /* no ack signal is sent on the following receiving byte */
-            dev->C1 |= I2C_C1_TXAK_MASK;
-        }
+        length--;
```

Comment:
I moved the length decrement up here so that the if block below is exactly the same as the if block at the beginning of the function (L217), for improved readability.